osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementServiceTest.java

```diff
@@ -552,4 +582,222 @@ void checkSsldPendingFirmwareUpdateReturnsTrueAndDeletesPendingUpdateWithMatchin
+		   	Arrays.asList(new FirmwareVersion(FirmwareModuleType.SECURITY, VERSION_1))
+		);
+
+    	this.firmwareManagementService.handleGetFirmwareVersionResponse(firmwareVersionDtos, ids, "messageType", 1, 
+    			ResponseMessageResultType.OK, null);
+
+    	verifyNoInteractions(this.webServiceResponseMessageSender);
+    	verify(this.ssldPendingFirmwareUpdateRepository).delete(any());
+    }
+
+    @Test
+    public void testHandleGetFirmwareVersionResponseNotOk() {
+    	final CorrelationIds ids = this.getCorrelationIds();
+    	final List<FirmwareVersionDto> versionsOnDevice = new ArrayList<>();
+    	
+    	this.firmwareManagementService.handleGetFirmwareVersionResponse(versionsOnDevice, ids, "messageType", 1,
+    			ResponseMessageResultType.NOT_OK, null);
+
+    	verify(this.webServiceResponseMessageSender).send(this.responseMessageCaptor.capture());
+
+    	final ResponseMessage responseMessage = this.responseMessageCaptor.getValue();
+
+    	assertThat(responseMessage.getResult()).isEqualTo(ResponseMessageResultType.NOT_OK);
+    	assertThat(responseMessage.getOsgpException().getMessage())
+                .isEqualTo("Exception occurred while getting device firmware version");
+    }
+
+    @Test
+    public void testHandleGetFirmwareVersionErrorNotNull() {
+        final CorrelationIds ids = this.getCorrelationIds();
+        final List<FirmwareVersionDto> versionsOnDevice = new ArrayList<>();
+
+        this.firmwareManagementService.handleGetFirmwareVersionResponse(versionsOnDevice, ids, "messageType", 1,
+            ResponseMessageResultType.OK, this.defaultException);
+
+        verify(this.webServiceResponseMessageSender).send(this.responseMessageCaptor.capture());
+        verify(this.ssldPendingFirmwareUpdateRepository, never()).delete(any());
+
+        final ResponseMessage responseMessage = this.responseMessageCaptor.getValue();
+        final ResponseMessage expectedResponseMessage = ResponseMessage.newResponseMessageBuilder()
+            .withIds(ids)
+            .withResult(ResponseMessageResultType.NOT_OK)
+            .withOsgpException(new TechnicalException("Exception occurred while getting device firmware version"))
+            .withMessagePriority(1)
+            .build();
+
+        assertThat(responseMessage).usingRecursiveComparison().ignoringFields("dataObject").isEqualTo(expectedResponseMessage);
+    }
+
+    @Test
+    public void testHandleGetFirmwareVersionWithPendingUpdateIsNull() {
+        final CorrelationIds ids = this.getCorrelationIds();
+        final List<FirmwareVersionDto> versionsOnDevice = new ArrayList<>();
+
+        when(this.ssldPendingFirmwareUpdateRepository.findByDeviceIdentification(any())).thenReturn(null);
+
+        assertThatThrownBy(() -> {
+            this.firmwareManagementService.handleGetFirmwareVersionResponse(versionsOnDevice, ids, "messageType", 1,
+                    ResponseMessageResultType.OK, null);
+        }).hasMessage(null);
```

Comment:
Check for the exception type using either .isInstanceOf in this construction or use assertThatExceptionOfType(<Exception.class>).isThrownBy(() -> {